lib/shared.bash

```diff
@@ -102,14 +102,30 @@ function build_image_override_file_with_version() {
     exit 1
   fi
 
+  cache_from_not_available=($(bc <<< "$version < 3.2"))
+
   printf "version: '%s'\n" "$version"
   printf "services:\n"
 
   shift
   while test ${#} -gt 0 ; do
     printf "  %s:\n" "$1"
     printf "    image: %s\n" "$2"
-    shift 2
+
+    if [[ -n "$3" ]] ; then
+      if [[ $cache_from_not_available -gt 0 ]] ; then
```

Comment:
I generally use a "not null" check: `if [[ -n $cache_from_not_available ]]` for bool-like flags.

Reply:
:+1: will change